src/sqlodbc/opensearch_communication.cpp

```diff
@@ -1048,15 +1071,38 @@ std::string OpenSearchCommunication::GetClusterName() {
 /**
  * @brief Sets URL endpoint for SQL plugin. On failure to
  * determine appropriate endpoint, value is set to SQL_ENDPOINT_ERROR_STR
- * 
+ *
  */
 void OpenSearchCommunication::SetSqlEndpoint() {
+
+    // Serverless Elasticsearch is not supported.
+    if (isServerless()) {
+        sql_endpoint = SQL_ENDPOINT_OPENSEARCH;
+        return;
+    }
+
     std::string distribution = GetServerDistribution();
     if (distribution.empty()) {
-        sql_endpoint = SQL_ENDPOINT_ERROR_STR;
-    } else if (distribution.compare("opensearch") == 0) {
-        sql_endpoint = "/_plugins/_sql";
+        sql_endpoint = SQL_ENDPOINT_ERROR;
+    } else if (distribution == "opensearch") {
+        sql_endpoint = SQL_ENDPOINT_OPENSEARCH;
     } else {
-        sql_endpoint = "/_opendistro/_sql";
+        sql_endpoint = SQL_ENDPOINT_ELASTICSEARCH;
     }
 }
+
+/**
+ * Returns whether this is connecting to an OpenSearch Serverless cluster.
+ */
+bool OpenSearchCommunication::isServerless() {
+
+    // Specified in DSN configuration.
+    const std::string& is_serverless = m_rt_opts.conn.is_serverless;
```

Comment:
Can we make the is_serverless field itself a boolean and evaluate it earlier? Rather than every time isServerless() is called? It looks like isServerless() is called during IssueRequest() which is used repeatedly (for executing queries and paging through results).

Reply:
Good call. Added `is_serverless` member variable and a corresponding `SetIsServerless` method, which is called during `EstablishConnection` - so only once.